viewer/packages/data-providers/src/image-360-data-providers/Cdf360ImageAnnotationProvider.test.ts

```diff
@@ -0,0 +1,188 @@
+    data: {
+      text: 'b',
+      textRegion: { xMin: 0, xMax: 0.1, yMin: 0, yMax: 0.1 },
+      assetRef: { id: 999 }
+    },
+    status: 'approved'
+  });
+
+  const matchingHybridAnnotation = createAnnotationModel({
+    annotatedResourceId: 30,
+    annotationType: 'images.InstanceLink',
+    data: {
+      text: 'hybrid annotation for file ' + ARBITRARY_FILE_ID,
+      textRegion: { xMin: 0, xMax: 0.1, yMin: 0, yMax: 0.1 },
+      instanceRef: {
+        externalId: 'instance-1',
+        space: 'space-1',
+        instanceType: 'node',
+        sources: []
+      }
+    },
+    status: 'approved'
+  });
+
+  const mockReverseLookup = jest.fn<CogniteClient['annotations']['reverseLookup']>();
+  const mockAnnotationList = jest.fn<CogniteClient['annotations']['list']>();
+
+  mockReverseLookup.mockImplementation(
+    (): CursorAndAsyncIterator<AnnotationsAssetRef> =>
+      createCursorAndAsyncIteratorMock({
+        items: [
+          {
+            id: ARBITRARY_FILE_ID
+          }
+        ]
+      })
+  );
+
+  mockAnnotationList.mockImplementation(
+    (): CursorAndAsyncIterator<AnnotationModel> =>
+      createCursorAndAsyncIteratorMock({
+        items: [
+          createAnnotationModel({
+            annotatedResourceId: 30,
+            annotationType: 'images.AssetLink',
+            data: {
+              text: 'annotation for file ' + ARBITRARY_FILE_ID,
+              textRegion: { xMin: 0, xMax: 0.1, yMin: 0, yMax: 0.1 },
+              assetRef: { id: 10 }
+            },
+            status: 'approved'
+          })
+        ]
+      })
+  );
+
+  const sdkMock = new Mock<CogniteClient>()
+    .setup(p => p.annotations.reverseLookup)
+    .returns(mockReverseLookup)
+    .setup(p => p.annotations.list)
+    .returns(mockAnnotationList)
+    .object();
+
+  // Minimal mock implementations for entity/revision/collection sufficient for provider logic
+  function createMockRevision(
+    fileId: number,
+    annotations: AnnotationModel[]
+  ): Partial<Image360RevisionEntity<ClassicDataSourceType>> {
+    const descriptor = new Mock<Image360Descriptor<ClassicDataSourceType>>()
+      .setup(d => d.faceDescriptors)
+      .returns([{ fileId, face: 'front', mimeType: 'image/jpeg' }])
+      .object();
+
+    const annotationObject1 = new Mock<ImageAnnotationObject<ClassicDataSourceType>>()
+      .setup(a => a.annotation)
+      .returns(annotations[0])
+      .object();
+
+    const annotationObject2 = new Mock<ImageAnnotationObject<ClassicDataSourceType>>()
+      .setup(a => a.annotation)
+      .returns(annotations[1])
+      .object();
+
+    const revision = new Mock<Image360RevisionEntity<ClassicDataSourceType>>();
+    revision.setup(r => r.identifier).returns('revision-' + fileId);
+    revision.setup(r => r.getDescriptors()).returns(descriptor);
+    revision.setup(r => r.getAnnotations()).returns(Promise.resolve([annotationObject1, annotationObject2]));
+    return revision.object();
```

Comment:
[Nitpick] Is there a reason we don't do `new Mock().setup(...).returns(..).setup(...).returns(...).object()` like above?